In [2]:
import pandas as pd
old_maping = pd.read_csv('old_map.csv')

In [3]:
# Creating a new column called new_name.

old_maping['new_name'] = old_maping['JOB_NAME'].apply(lambda x: x[15:]if x[:15] == 'm_STG_DWH_LOAD_'
                                                      else (x[11:] if x[:11] == 'm_DWH_LOAD_' else x))

In [4]:
# I have add a new column called 'USED_TOOL' so that import the new maping with value USED_TOOL = 'INFORMATICA'
old_maping = old_maping[old_maping.USED_TOOL == 'SSIS']
old_maping.head()

WORK_STRAM                        JOB_NAME JOB_INIT_DATE FREQUENCY  \
0        STG         m_STG_DWH_LOAD_ZDT_ERST     04-JUL-18     Daily   
1        STG         m_STG_DWH_LOAD_ZDT_ERTD     04-JUL-18     Daily   
2        STG   m_STG_DWH_LOAD_ZDT_MCI_ERRLOG     09-MAY-18     Daily   
3        STG     m_STG_DWH_LOAD_ZDT_OBJ_CASE     25-DEC-18     Daily   
4        STG  m_STG_DWH_LOAD_ZDT_SADAD_CHECK     02-DEC-18     Daily   

  USED_TOOL         new_name  
0      SSIS         ZDT_ERST  
1      SSIS         ZDT_ERTD  
2      SSIS   ZDT_MCI_ERRLOG  
3      SSIS     ZDT_OBJ_CASE  
4      SSIS  ZDT_SADAD_CHECK

In [5]:
# We have 503 row in the old mapping.
old_maping.count()

WORK_STRAM       503
JOB_NAME         503
JOB_INIT_DATE    503
FREQUENCY        503
USED_TOOL        503
new_name         503
dtype: int64

# Informatica log for STG Maps. (To get the STG Tables)

In [6]:
info_log_STG = pd.read_csv('info_STG_MAP.csv')
info_log_STG

WORKFLOW_NAME    OWNER SUBJECT_AREA                  SESSION_NAME  \
0      wf_TRM_STG  TRM_STG      TRM_STG      s_m_STG_TRM_TRANSACTIONS   
1      wf_TRM_STG  TRM_STG      TRM_STG  s_m_STG_TRM_ALL_TRANSACTIONS   
2      wf_TRM_STG  TRM_STG      TRM_STG              s_m_STG_TRM_BNKA   
3      wf_TRM_STG  TRM_STG      TRM_STG          s_m_STG_TRM_ZDT_VTTD   
4      wf_TRM_STG  TRM_STG      TRM_STG            s_m_STG_TRM_BUT000   
..            ...      ...          ...                           ...   
247    wf_TRM_STG  TRM_STG      TRM_STG   s_m_STG_TRM_ZDT_SADAD_CHECK   
248    wf_TRM_STG  TRM_STG      TRM_STG      s_m_STG_TRM_ZDT_APPL_LOG   
249    wf_TRM_STG  TRM_STG      TRM_STG            s_m_STG_TRM_FKKVKP   
250    wf_TRM_STG  TRM_STG      TRM_STG            s_m_STG_TRM_BUT050   
251    wf_TRM_STG  TRM_STG      TRM_STG          s_m_GenerateParmFile   

     SOURCE_ROW_COUNT  TARGET_ROW_COUNT JOB_START_DATE JOB_END_DATE   STATUS  
0            27929955          27929955      02-APR-20    02-APR-20  Success  
1            30753981          30753981      02-APR-20    02-APR-20  Success  
2                 136               136      02-APR-20    02-APR-20  Success  
3              295237            295237      02-APR-20    02-APR-20  Success  
4             1132308           1132308      02-APR-20    02-APR-20  Success  
..                ...               ...            ...          ...      ...  
247             14353             14353      02-APR-20    02-APR-20  Success  
248          10539451          10539451      02-APR-20    02-APR-20  Success  
249           2237915           2237915      02-APR-20    02-APR-20  Success  
250            271452            271452      02-APR-20    02-APR-20  Success  
251               434               434      02-APR-20    02-APR-20  Success  

[252 rows x 9 columns]

In [7]:
# I am intersted in teh column SESSION_NAME so I will extract it.
info_log_STG = info_log_STG.iloc[: , [3]]
info_log_STG

SESSION_NAME
0        s_m_STG_TRM_TRANSACTIONS
1    s_m_STG_TRM_ALL_TRANSACTIONS
2                s_m_STG_TRM_BNKA
3            s_m_STG_TRM_ZDT_VTTD
4              s_m_STG_TRM_BUT000
..                            ...
247   s_m_STG_TRM_ZDT_SADAD_CHECK
248      s_m_STG_TRM_ZDT_APPL_LOG
249            s_m_STG_TRM_FKKVKP
250            s_m_STG_TRM_BUT050
251          s_m_GenerateParmFile

[252 rows x 1 columns]

In [9]:
# There is 252 STG map on the informatica log.
info_log_STG.count()

SESSION_NAME    252
dtype: int64

In [10]:
# Adding a new column called 'new_name'
info_log_STG['new_name'] = info_log_STG['SESSION_NAME'].apply(lambda x: x[12:] if x[:12] == 's_m_STG_TRM_' 
                                                      else x)

/Users/ahmedzelikha/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [11]:
info_log_STG

SESSION_NAME              new_name
0        s_m_STG_TRM_TRANSACTIONS          TRANSACTIONS
1    s_m_STG_TRM_ALL_TRANSACTIONS      ALL_TRANSACTIONS
2                s_m_STG_TRM_BNKA                  BNKA
3            s_m_STG_TRM_ZDT_VTTD              ZDT_VTTD
4              s_m_STG_TRM_BUT000                BUT000
..                            ...                   ...
247   s_m_STG_TRM_ZDT_SADAD_CHECK       ZDT_SADAD_CHECK
248      s_m_STG_TRM_ZDT_APPL_LOG          ZDT_APPL_LOG
249            s_m_STG_TRM_FKKVKP                FKKVKP
250            s_m_STG_TRM_BUT050                BUT050
251          s_m_GenerateParmFile  s_m_GenerateParmFile

[252 rows x 2 columns]

# Informatica log for EWD Maps. (To get the EDW Tables)

In [12]:
info_log_EWH = pd.read_csv('Target.csv')

In [13]:
# Adding a new column called 'new_name'
info_log_EWH['new_name'] = info_log_EWH['SESSION_NAME'].apply(lambda x: x[8:] if x[:8] == 's_m_STG_' 
                                                      else x)

In [14]:
# I don't need the GSTG Maps
info_log_EWH = info_log_EWH[info_log_EWH["SESSION_NAME"].str[:8] == 's_m_DWH_']
info_log_EWH = info_log_EWH[info_log_EWH["SESSION_NAME"].str[:12] !='s_m_DWH_GSTC']
info_log_EWH

SESSION_NAME                           new_name
4                  s_m_DWH_QWAEEM_FILES               s_m_DWH_QWAEEM_FILES
95          s_m_DWH_BP_DATA_SOURCE_DESC        s_m_DWH_BP_DATA_SOURCE_DESC
96          s_m_DWH_CMS_CATEGORIES_DESC        s_m_DWH_CMS_CATEGORIES_DESC
97        s_m_DWH_BP_ADDRESS_TYPES_DESC      s_m_DWH_BP_ADDRESS_TYPES_DESC
98     s_m_DWH_VT_AMENDMENT_REASON_DESC   s_m_DWH_VT_AMENDMENT_REASON_DESC
...                                 ...                                ...
1023             s_m_DWH_RETURN_DETAILS             s_m_DWH_RETURN_DETAILS
1026                 s_m_DWH_ICR_MASTER                 s_m_DWH_ICR_MASTER
1027           s_m_DWH_CO_FI_LINE_ITEMS           s_m_DWH_CO_FI_LINE_ITEMS
1028       s_m_DWH_CRM_AUDIT_REQUEST_DM       s_m_DWH_CRM_AUDIT_REQUEST_DM
1033  s_m_DWH_CO_BUSINESS_TRANSACT_DESC  s_m_DWH_CO_BUSINESS_TRANSACT_DESC

[250 rows x 2 columns]

In [15]:
# Get the EDW tables names.
info_log_EWH['new_name'] = info_log_EWH['SESSION_NAME'].apply(lambda x: x[4:] if x[:4] == 's_m_' 
                                                      else x)
info_log_EWH

SESSION_NAME                       new_name
4                  s_m_DWH_QWAEEM_FILES               DWH_QWAEEM_FILES
95          s_m_DWH_BP_DATA_SOURCE_DESC        DWH_BP_DATA_SOURCE_DESC
96          s_m_DWH_CMS_CATEGORIES_DESC        DWH_CMS_CATEGORIES_DESC
97        s_m_DWH_BP_ADDRESS_TYPES_DESC      DWH_BP_ADDRESS_TYPES_DESC
98     s_m_DWH_VT_AMENDMENT_REASON_DESC   DWH_VT_AMENDMENT_REASON_DESC
...                                 ...                            ...
1023             s_m_DWH_RETURN_DETAILS             DWH_RETURN_DETAILS
1026                 s_m_DWH_ICR_MASTER                 DWH_ICR_MASTER
1027           s_m_DWH_CO_FI_LINE_ITEMS           DWH_CO_FI_LINE_ITEMS
1028       s_m_DWH_CRM_AUDIT_REQUEST_DM       DWH_CRM_AUDIT_REQUEST_DM
1033  s_m_DWH_CO_BUSINESS_TRANSACT_DESC  DWH_CO_BUSINESS_TRANSACT_DESC

[250 rows x 2 columns]

# joining the DataFrames (Old_mapping with staging maps in informatica)

In [19]:
left_join = pd.merge(old_maping[old_maping["WORK_STRAM"] == 'STG'], info_log_STG, on='new_name', how='left')
left_join

WORK_STRAM                         JOB_NAME JOB_INIT_DATE FREQUENCY  \
0          STG          m_STG_DWH_LOAD_ZDT_ERST     04-JUL-18     Daily   
1          STG          m_STG_DWH_LOAD_ZDT_ERTD     04-JUL-18     Daily   
2          STG    m_STG_DWH_LOAD_ZDT_MCI_ERRLOG     09-MAY-18     Daily   
3          STG      m_STG_DWH_LOAD_ZDT_OBJ_CASE     25-DEC-18     Daily   
4          STG   m_STG_DWH_LOAD_ZDT_SADAD_CHECK     02-DEC-18     Daily   
..         ...                              ...           ...       ...   
245        STG   m_STG_DWH_LOAD_ZDT_CASE_OBJECT     29-NOV-18     Daily   
246        STG  m_STG_DWH_LOAD_ZDT_CUST_COUNTRY     16-SEP-18     Daily   
247        STG          m_STG_DWH_LOAD_ZDT_ERAT     04-JUL-18     Daily   
248        STG          m_STG_DWH_LOAD_ZDT_ERCR     04-JUL-18     Daily   
249        STG          m_STG_DWH_LOAD_ZDT_ERGT     04-JUL-18     Daily   

    USED_TOOL          new_name                  SESSION_NAME  
0        SSIS          ZDT_ERST          s_m_STG_TRM_ZDT_ERST  
1        SSIS          ZDT_ERTD          s_m_STG_TRM_ZDT_ERTD  
2        SSIS    ZDT_MCI_ERRLOG    s_m_STG_TRM_ZDT_MCI_ERRLOG  
3        SSIS      ZDT_OBJ_CASE      s_m_STG_TRM_ZDT_OBJ_CASE  
4        SSIS   ZDT_SADAD_CHECK   s_m_STG_TRM_ZDT_SADAD_CHECK  
..        ...               ...                           ...  
245      SSIS   ZDT_CASE_OBJECT   s_m_STG_TRM_ZDT_CASE_OBJECT  
246      SSIS  ZDT_CUST_COUNTRY  s_m_STG_TRM_ZDT_CUST_COUNTRY  
247      SSIS          ZDT_ERAT          s_m_STG_TRM_ZDT_ERAT  
248      SSIS          ZDT_ERCR          s_m_STG_TRM_ZDT_ERCR  
249      SSIS          ZDT_ERGT          s_m_STG_TRM_ZDT_ERGT  

[250 rows x 7 columns]

In [18]:
left_join.count()

WORK_STRAM       250
JOB_NAME         250
JOB_INIT_DATE    250
FREQUENCY        250
USED_TOOL        250
new_name         250
SESSION_NAME     242
dtype: int64

# joining the DataFrames (Old_mapping with EDW maps in informatica)

In [20]:
left_join_T = pd.merge(old_maping[old_maping["WORK_STRAM"] == 'DWH'], info_log_EWH, on='new_name', how='left')
left_join_T

WORK_STRAM                                 JOB_NAME JOB_INIT_DATE  \
0          DWH              m_DWH_LOAD_DWH_BP_ZI11_LIST     26-DEC-19   
1          DWH       m_DWH_LOAD_DWH_RETURN_FZ12_DETAILS     25-DEC-19   
2          DWH       m_DWH_LOAD_DWH_RETURN_ZI10_DETAILS     25-DEC-19   
3          DWH       m_DWH_LOAD_DWH_RETURN_ZI11_DETAILS     25-DEC-19   
4          DWH               m_DWH_LOAD_DWH_SADAD_RECON     30-JAN-20   
..         ...                                      ...           ...   
175        DWH        m_DWH_LOAD_DWH_WH_ADDRESS_DETAILS     04-JUL-18   
176        DWH           m_DWH_LOAD_DWH_WH_BANK_DETAILS     04-JUL-18   
177        DWH          m_DWH_LOAD_DWH_WH_GOODS_DETAILS     04-JUL-18   
178        DWH         m_DWH_LOAD_DWH_WH_HEADER_DETAILS     04-JUL-18   
179        DWH  m_DWH_LOAD_DWH_WH_OWNER_MANGERS_DETAILS     04-JUL-18   

    FREQUENCY USED_TOOL                      new_name  \
0       Daily      SSIS              DWH_BP_ZI11_LIST   
1       Daily      SSIS       DWH_RETURN_FZ12_DETAILS   
2       Daily      SSIS       DWH_RETURN_ZI10_DETAILS   
3       Daily      SSIS       DWH_RETURN_ZI11_DETAILS   
4       Daily      SSIS               DWH_SADAD_RECON   
..        ...       ...                           ...   
175     Daily      SSIS        DWH_WH_ADDRESS_DETAILS   
176     Daily      SSIS           DWH_WH_BANK_DETAILS   
177     Daily      SSIS          DWH_WH_GOODS_DETAILS   
178     Daily      SSIS         DWH_WH_HEADER_DETAILS   
179     Daily      SSIS  DWH_WH_OWNER_MANGERS_DETAILS   

                         SESSION_NAME  
0                s_m_DWH_BP_ZI11_LIST  
1         s_m_DWH_RETURN_FZ12_DETAILS  
2         s_m_DWH_RETURN_ZI10_DETAILS  
3         s_m_DWH_RETURN_ZI11_DETAILS  
4                 s_m_DWH_SADAD_RECON  
..                                ...  
175        s_m_DWH_WH_ADDRESS_DETAILS  
176           s_m_DWH_WH_BANK_DETAILS  
177          s_m_DWH_WH_GOODS_DETAILS  
178         s_m_DWH_WH_HEADER_DETAILS  
179  s_m_DWH_WH_OWNER_MANGERS_DETAILS  

[180 rows x 7 columns]

In [21]:
left_join_T.count()

WORK_STRAM       180
JOB_NAME         180
JOB_INIT_DATE    180
FREQUENCY        180
USED_TOOL        180
new_name         180
SESSION_NAME     170
dtype: int64

In [26]:
# Combine the two dataframes (STG Tables with EDW Tables)

com = pd.concat([left_join, left_join_T])
com['USED_TOOL'] = 'INFORMATICA'
com

WORK_STRAM                                 JOB_NAME JOB_INIT_DATE  \
0          STG                  m_STG_DWH_LOAD_ZDT_ERST     04-JUL-18   
1          STG                  m_STG_DWH_LOAD_ZDT_ERTD     04-JUL-18   
2          STG            m_STG_DWH_LOAD_ZDT_MCI_ERRLOG     09-MAY-18   
3          STG              m_STG_DWH_LOAD_ZDT_OBJ_CASE     25-DEC-18   
4          STG           m_STG_DWH_LOAD_ZDT_SADAD_CHECK     02-DEC-18   
..         ...                                      ...           ...   
175        DWH        m_DWH_LOAD_DWH_WH_ADDRESS_DETAILS     04-JUL-18   
176        DWH           m_DWH_LOAD_DWH_WH_BANK_DETAILS     04-JUL-18   
177        DWH          m_DWH_LOAD_DWH_WH_GOODS_DETAILS     04-JUL-18   
178        DWH         m_DWH_LOAD_DWH_WH_HEADER_DETAILS     04-JUL-18   
179        DWH  m_DWH_LOAD_DWH_WH_OWNER_MANGERS_DETAILS     04-JUL-18   

    FREQUENCY    USED_TOOL                      new_name  \
0       Daily  INFORMATICA                      ZDT_ERST   
1       Daily  INFORMATICA                      ZDT_ERTD   
2       Daily  INFORMATICA                ZDT_MCI_ERRLOG   
3       Daily  INFORMATICA                  ZDT_OBJ_CASE   
4       Daily  INFORMATICA               ZDT_SADAD_CHECK   
..        ...          ...                           ...   
175     Daily  INFORMATICA        DWH_WH_ADDRESS_DETAILS   
176     Daily  INFORMATICA           DWH_WH_BANK_DETAILS   
177     Daily  INFORMATICA          DWH_WH_GOODS_DETAILS   
178     Daily  INFORMATICA         DWH_WH_HEADER_DETAILS   
179     Daily  INFORMATICA  DWH_WH_OWNER_MANGERS_DETAILS   

                         SESSION_NAME  
0                s_m_STG_TRM_ZDT_ERST  
1                s_m_STG_TRM_ZDT_ERTD  
2          s_m_STG_TRM_ZDT_MCI_ERRLOG  
3            s_m_STG_TRM_ZDT_OBJ_CASE  
4         s_m_STG_TRM_ZDT_SADAD_CHECK  
..                                ...  
175        s_m_DWH_WH_ADDRESS_DETAILS  
176           s_m_DWH_WH_BANK_DETAILS  
177          s_m_DWH_WH_GOODS_DETAILS  
178         s_m_DWH_WH_HEADER_DETAILS  
179  s_m_DWH_WH_OWNER_MANGERS_DETAILS  

[430 rows x 7 columns]

In [27]:
# There is Tables the not found in the revamp maps as the following
Col_not_match = com[pd.isnull(com["SESSION_NAME"])]
Col_not_match

WORK_STRAM                                     JOB_NAME JOB_INIT_DATE  \
30         STG                     m_STG_DWH_LOAD_FKK_SEC_N     24-FEB-20   
31         STG              m_STG_DWH_LOAD_TFK_SEC_STATUS_T     20-FEB-20   
65         STG               m_STG_DWH_LOAD_DFMCA_FORM_VERS     27-MAY-19   
68         STG          m_STG_DWH_LOAD_EXT_CONTRACTS_DETAIL     21-OCT-18   
98         STG          m_STG_DWH_LOAD_STG_BP_MAIN_ACTIVITY     04-JUL-18   
99         STG      m_STG_DWH_LOAD_STG_TRM_ALL_TRANSACTIONS     22-NOV-18   
100        STG          m_STG_DWH_LOAD_STG_TRM_TRANSACTIONS     22-NOV-18   
165        STG                   m_STG_DWH_LOAD_WEB_IMP_SAP     16-OCT-18   
6          DWH    m_DWH_LOAD_DWH_CMS_REV_VT_SEC_STATUS_DESC     20-FEB-20   
7          DWH                        m_DWH_CUST_PORTS_DESC     03-OCT-18   
83         DWH              m_DWH_LOAD_DWH_CORR_STSTUS_DESC     17-DEC-18   
86         DWH               m_DWH_LOAD_DWH_CUST_CALL_ORDER     16-SEP-18   
88         DWH  m_DWH_LOAD_DWH_CUST_DECLARATION_HEADER_CULG     13-SEP-18   
89         DWH   m_DWH_LOAD_DWH_CUST_DECLARATION_ITEMS_CULG     13-SEP-18   
133        DWH      m_DWH_LOAD_DWH_INSTALLMENT_PLANS_HEADER     30-DEC-18   
134        DWH       m_DWH_LOAD_DWH_INSTALLMENT_PLANS_ITEMS     30-DEC-18   
172        DWH      m_DWH_LOAD_DWH_VT_Amendment_Reason_Desc     04-JUL-18   
173        DWH          m_DWH_LOAD_DWH_VT_DeReg_REASON_DESC     07-OCT-18   

    FREQUENCY    USED_TOOL                          new_name SESSION_NAME  
30      Daily  INFORMATICA                         FKK_SEC_N          NaN  
31      Daily  INFORMATICA                  TFK_SEC_STATUS_T          NaN  
65      Daily  INFORMATICA                   DFMCA_FORM_VERS          NaN  
68      Daily  INFORMATICA              EXT_CONTRACTS_DETAIL          NaN  
98      Daily  INFORMATICA              STG_BP_MAIN_ACTIVITY          NaN  
99      Daily  INFORMATICA          STG_TRM_ALL_TRANSACTIONS          NaN  
100     Daily  INFORMATICA              STG_TRM_TRANSACTIONS          NaN  
165     Daily  INFORMATICA                       WEB_IMP_SAP          NaN  
6       Daily  INFORMATICA    DWH_CMS_REV_VT_SEC_STATUS_DESC          NaN  
7       Daily  INFORMATICA             m_DWH_CUST_PORTS_DESC          NaN  
83      Daily  INFORMATICA              DWH_CORR_STSTUS_DESC          NaN  
86      Daily  INFORMATICA               DWH_CUST_CALL_ORDER          NaN  
88      Daily  INFORMATICA  DWH_CUST_DECLARATION_HEADER_CULG          NaN  
89      Daily  INFORMATICA   DWH_CUST_DECLARATION_ITEMS_CULG          NaN  
133     Daily  INFORMATICA      DWH_INSTALLMENT_PLANS_HEADER          NaN  
134     Daily  INFORMATICA       DWH_INSTALLMENT_PLANS_ITEMS          NaN  
172     Daily  INFORMATICA      DWH_VT_Amendment_Reason_Desc          NaN  
173     Daily  INFORMATICA          DWH_VT_DeReg_REASON_DESC          NaN

# The Final result for mapping 

<B>430</B> table
<B>49</B> datamart
<B>total = 479</B>

the diffrence is 24 (I think they will be the GSTC maps)

In [35]:
# the results.
result = com.iloc[:, [0, 6, 2, 3, 4]]
result

WORK_STRAM                      SESSION_NAME JOB_INIT_DATE FREQUENCY  \
0          STG              s_m_STG_TRM_ZDT_ERST     04-JUL-18     Daily   
1          STG              s_m_STG_TRM_ZDT_ERTD     04-JUL-18     Daily   
2          STG        s_m_STG_TRM_ZDT_MCI_ERRLOG     09-MAY-18     Daily   
3          STG          s_m_STG_TRM_ZDT_OBJ_CASE     25-DEC-18     Daily   
4          STG       s_m_STG_TRM_ZDT_SADAD_CHECK     02-DEC-18     Daily   
..         ...                               ...           ...       ...   
175        DWH        s_m_DWH_WH_ADDRESS_DETAILS     04-JUL-18     Daily   
176        DWH           s_m_DWH_WH_BANK_DETAILS     04-JUL-18     Daily   
177        DWH          s_m_DWH_WH_GOODS_DETAILS     04-JUL-18     Daily   
178        DWH         s_m_DWH_WH_HEADER_DETAILS     04-JUL-18     Daily   
179        DWH  s_m_DWH_WH_OWNER_MANGERS_DETAILS     04-JUL-18     Daily   

       USED_TOOL  
0    INFORMATICA  
1    INFORMATICA  
2    INFORMATICA  
3    INFORMATICA  
4    INFORMATICA  
..           ...  
175  INFORMATICA  
176  INFORMATICA  
177  INFORMATICA  
178  INFORMATICA  
179  INFORMATICA  

[430 rows x 5 columns]

In [32]:
result.to_csv('mapping_results.csv')